components/scons/scons.p5m

```diff
@@ -0,0 +1,442 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2016 Jim Klimov
+#
+
+<transform file path=usr.*/man/.+ -> default mangler.man.stability uncommitted>
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+# cd proto && ( find . -type f | sed 's,^./,file path=,' ; find . -type l -ls | sed -e 's,^.*\./,link path=,' -e 's, -> , target=,' ) >> p5m
+file path=usr/share/man/man1/sconsign.1
+file path=usr/share/man/man1/scons-time.1
+file path=usr/share/man/man1/scons.1
+file path=usr/bin/scons-$(COMPONENT_VERSION).py
+file path=usr/bin/scons-$(COMPONENT_VERSION).pyc
+link path=usr/bin/scons-$(COMPONENT_VERSION) target=scons-$(COMPONENT_VERSION).py
+link path=usr/bin/scons target=scons-$(COMPONENT_VERSION)
+file path=usr/bin/scons-time-$(COMPONENT_VERSION).py
+file path=usr/bin/scons-time-$(COMPONENT_VERSION).pyc
+link path=usr/bin/scons-time-$(COMPONENT_VERSION) target=scons-time-$(COMPONENT_VERSION).py
```

Comment:
Originally they do deliver .py content as just 'scons'; I thought it can
help to have precompiled bins there.

They have options to deliver versioned or plain bins and dirs; I have no
idea if we'd need several at once (hope not, but several projects suggest
using scons-2.3.0 to build them).

Can fix site/vendor dir, I hope... such path should not compiled in
anywhere? - would it suffice that it is in a python-controlled area?

--
Typos courtesy of GMail on my Samsung Android
21.02.2016 21:43 пользователь "Alexander Pyhalov" <notifications@github.com>
написал:

> In components/scons/scons.p5m
> <https://github.com/OpenIndiana/oi-userland/pull/1709#discussion_r53572974>
> :
>
> > +set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
> > +set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
> > +
> > +license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
> > +
> > +# cd proto && ( find . -type f | sed 's,^./,file path=,' ; find . -type l -ls | sed -e 's,^.*\./,link path=,' -e 's, -> , target=,' ) >> p5m
> > +file path=usr/share/man/man1/sconsign.1
> > +file path=usr/share/man/man1/scons-time.1
> > +file path=usr/share/man/man1/scons.1
> > +file path=usr/bin/scons-$(COMPONENT_VERSION).py
> > +file path=usr/bin/scons-$(COMPONENT_VERSION).pyc
> > +link path=usr/bin/scons-$(COMPONENT_VERSION) target=scons-$(COMPONENT_VERSION).py
> > +link path=usr/bin/scons target=scons-$(COMPONENT_VERSION)
> > +file path=usr/bin/scons-time-$(COMPONENT_VERSION).py
> > +file path=usr/bin/scons-time-$(COMPONENT_VERSION).pyc
> > +link path=usr/bin/scons-time-$(COMPONENT_VERSION) target=scons-time-$(COMPONENT_VERSION).py
>
> Can we avoid delivering py and pyc files in /usr/bin ? At last it can be
> scons.py delivered as scons.
> Do we want to deliver multiple scons versions simultaneously? If not, why
> should we deliver scons-$(COMPONENT_VERSION) instead of just scons?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/pull/1709/files#r53572974>.
>